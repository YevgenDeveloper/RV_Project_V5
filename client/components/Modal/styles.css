@import 'client/styles/variables';

.modal {
  z-index: 2000;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.fade {
  z-index: 9001;
  cursor: pointer;
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background-color: $color-fade;
  animation: fade 0.3s;
}

.close {
  z-index: 9003;
  cursor: pointer;
  width: 44px;
  height: 44px;
  border-radius: 22px;
  position: absolute;
  top: 0;
  right: 0;
  background: white url('client/assets/icn-close.svg') 50% 50% no-repeat;
  background-size: 14px;
  opacity: 0.3;
  &:hover {
    opacity: 1;
  }
}

.body {
  z-index: 9002;
  position: absolute;
  top: 50%;
  left: 50%;
  min-width: 360px;
  max-width: 700px;
  max-height: 90%;
  overflow: hidden;
  border-radius: 8px;
  background-color: #fff;
  box-shadow: 0 20px 60px -5px rgba(0, 0, 0, 0.6);
  transform-origin: 50% 50%;
  transform: translate(-50%, -50%) scale(1);
  transition-property: transform, opacity;
  transition-delay: 0.2s;
  animation: body 0.3s cubic-bezier(0.23, 1, 0.32, 1);
}

.content {
  z-index: 1;
  padding: 30px;
  color: $color-text-default;

  :global h1 {
    font-size: 24px;
    margin: 0 0 1em;
    user-select: text;
  }

  :global p {
    font-size: $font-size-default;
    margin: 0 0 1em;
    user-select: text;
  }
}

@keyframes fade {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@keyframes body {
  0% {
    opacity: 0;
    transform: translate(-50%, -50%) scale(1.05);
  }

  100% {
    opacity: 1;
    transform: translate(-50%, -50%) scale(1);
  }
}
