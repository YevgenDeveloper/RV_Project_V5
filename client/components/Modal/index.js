// @flow

import type { Node } from 'react';

import React, { Component } from 'react';
import classNames from 'classnames/bind';
import { Scrollbars } from 'react-custom-scrollbars';

import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  className?: string,
  close: void => void,
  children?: Node,
};

export default class Modal extends Component<Props, {}> {
  static defaultProps = {
    className: undefined,
    children: null,
  };

  componentDidMount() {
    const { body } = window.document;

    if (body) {
      body.style.overflow = 'hidden';
    }
  }

  componentWillUnmount() {
    const { body } = window.document;

    if (body) {
      body.style.overflow = 'auto';
    }
  }

  render() {
    return (
      <div className={cx(styles.modal, this.props.className)}>
        <button
          type="button"
          className={styles.fade}
          onClick={() => this.props.close()}
          title="Close modal"
          aria-label="Close modal"
        />
        <div className={styles.body}>
          <button
            type="button"
            className={styles.close}
            onClick={() => this.props.close()}
            title="Close modal"
            aria-label="Close modal"
          />
          <Scrollbars
            autoHeight
            autoHeightMin={80}
            autoHeightMax="calc(100vh - 120px)"
          >
            <div className={styles.content}>{this.props.children}</div>
          </Scrollbars>
        </div>
      </div>
    );
  }
}
