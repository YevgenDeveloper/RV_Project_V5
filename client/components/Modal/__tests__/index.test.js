// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import Modal from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(<Modal />);

  expect(wrapper).toMatchSnapshot();
});
