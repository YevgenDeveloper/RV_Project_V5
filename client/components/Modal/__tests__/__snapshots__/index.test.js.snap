// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          aria-label="Close modal"
          className="fade"
          onClick={[Function]}
          title="Close modal"
          type="button"
        />,
        <div
          className="body"
        >
          <button
            aria-label="Close modal"
            className="close"
            onClick={[Function]}
            title="Close modal"
            type="button"
          />
          <Scrollbars
            autoHeight={true}
            autoHeightMax="calc(100vh - 120px)"
            autoHeightMin={80}
            autoHide={false}
            autoHideDuration={200}
            autoHideTimeout={1000}
            hideTracksWhenNotNeeded={false}
            renderThumbHorizontal={[Function]}
            renderThumbVertical={[Function]}
            renderTrackHorizontal={[Function]}
            renderTrackVertical={[Function]}
            renderView={[Function]}
            tagName="div"
            thumbMinSize={30}
            universal={false}
          >
            <div
              className="content"
            />
          </Scrollbars>
        </div>,
      ],
      "className": "modal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Close modal",
          "className": "fade",
          "onClick": [Function],
          "title": "Close modal",
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              aria-label="Close modal"
              className="close"
              onClick={[Function]}
              title="Close modal"
              type="button"
            />,
            <Scrollbars
              autoHeight={true}
              autoHeightMax="calc(100vh - 120px)"
              autoHeightMin={80}
              autoHide={false}
              autoHideDuration={200}
              autoHideTimeout={1000}
              hideTracksWhenNotNeeded={false}
              renderThumbHorizontal={[Function]}
              renderThumbVertical={[Function]}
              renderTrackHorizontal={[Function]}
              renderTrackVertical={[Function]}
              renderView={[Function]}
              tagName="div"
              thumbMinSize={30}
              universal={false}
            >
              <div
                className="content"
              />
            </Scrollbars>,
          ],
          "className": "body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Close modal",
              "className": "close",
              "onClick": [Function],
              "title": "Close modal",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoHeight": true,
              "autoHeightMax": "calc(100vh - 120px)",
              "autoHeightMin": 80,
              "autoHide": false,
              "autoHideDuration": 200,
              "autoHideTimeout": 1000,
              "children": <div
                className="content"
              />,
              "hideTracksWhenNotNeeded": false,
              "renderThumbHorizontal": [Function],
              "renderThumbVertical": [Function],
              "renderTrackHorizontal": [Function],
              "renderTrackVertical": [Function],
              "renderView": [Function],
              "tagName": "div",
              "thumbMinSize": 30,
              "universal": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "content",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            aria-label="Close modal"
            className="fade"
            onClick={[Function]}
            title="Close modal"
            type="button"
          />,
          <div
            className="body"
          >
            <button
              aria-label="Close modal"
              className="close"
              onClick={[Function]}
              title="Close modal"
              type="button"
            />
            <Scrollbars
              autoHeight={true}
              autoHeightMax="calc(100vh - 120px)"
              autoHeightMin={80}
              autoHide={false}
              autoHideDuration={200}
              autoHideTimeout={1000}
              hideTracksWhenNotNeeded={false}
              renderThumbHorizontal={[Function]}
              renderThumbVertical={[Function]}
              renderTrackHorizontal={[Function]}
              renderTrackVertical={[Function]}
              renderView={[Function]}
              tagName="div"
              thumbMinSize={30}
              universal={false}
            >
              <div
                className="content"
              />
            </Scrollbars>
          </div>,
        ],
        "className": "modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Close modal",
            "className": "fade",
            "onClick": [Function],
            "title": "Close modal",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                aria-label="Close modal"
                className="close"
                onClick={[Function]}
                title="Close modal"
                type="button"
              />,
              <Scrollbars
                autoHeight={true}
                autoHeightMax="calc(100vh - 120px)"
                autoHeightMin={80}
                autoHide={false}
                autoHideDuration={200}
                autoHideTimeout={1000}
                hideTracksWhenNotNeeded={false}
                renderThumbHorizontal={[Function]}
                renderThumbVertical={[Function]}
                renderTrackHorizontal={[Function]}
                renderTrackVertical={[Function]}
                renderView={[Function]}
                tagName="div"
                thumbMinSize={30}
                universal={false}
              >
                <div
                  className="content"
                />
              </Scrollbars>,
            ],
            "className": "body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Close modal",
                "className": "close",
                "onClick": [Function],
                "title": "Close modal",
                "type": "button",
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoHeight": true,
                "autoHeightMax": "calc(100vh - 120px)",
                "autoHeightMin": 80,
                "autoHide": false,
                "autoHideDuration": 200,
                "autoHideTimeout": 1000,
                "children": <div
                  className="content"
                />,
                "hideTracksWhenNotNeeded": false,
                "renderThumbHorizontal": [Function],
                "renderThumbVertical": [Function],
                "renderTrackHorizontal": [Function],
                "renderTrackVertical": [Function],
                "renderView": [Function],
                "tagName": "div",
                "thumbMinSize": 30,
                "universal": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "content",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
